src/openms/include/OpenMS/FILTERING/TRANSFORMERS/SpectraMerger.h

```diff
@@ -278,6 +284,240 @@ namespace OpenMS
+        Size idx = spectra_to_average_over.begin()->first;    // index of first spectrum to be averaged
+        SpectrumSettings::SpectrumType spectrum_type = exp[idx].getType();
+        if (spectrum_type == SpectrumSettings::UNKNOWN)
+        {
+          spectrum_type = PeakTypeEstimator().estimateType(exp[idx].begin(), exp[idx].end());
+        }
+        
+        if (spectrum_type == SpectrumSettings::PEAKS)
+        {
+          averageCentroidSpectra_(exp, spectra_to_average_over, *it_mslevel);
+        }
+        else
+        {
+          averageProfileSpectra_(exp, spectra_to_average_over, *it_mslevel);
+        }
+      }
+        
+      exp.sortSpectra();
+      
+      return;
+    }
+
+    /**
+     * @brief Gaussian average over neighbouring spectra
+     *
+     * @param exp   experimental data to be averaged
+     */
+    template <typename MapType>
+    void averageGaussian(MapType & exp)
+    {
+      IntList ms_levels = param_.getValue("average_Gaussian:ms_levels");
+      double fwhm(param_.getValue("average_Gaussian:rt_FWHM"));
+      double factor = -4*log(2)/(fwhm*fwhm);    // numerical factor within Gaussian
+      double cutoff(param_.getValue("average_Gaussian:cutoff"));
+
+      // loop over MS levels
+      for (IntList::iterator it_mslevel = ms_levels.begin(); it_mslevel < ms_levels.end(); ++it_mslevel)
+      {
+        AverageBlocks spectra_to_average_over;
+        
+        // loop over RT
+        int n(0);    // spectrum index
+        for (typename MapType::const_iterator it_rt = exp.begin(); it_rt != exp.end(); ++it_rt)
+        {
+          if (Int(it_rt->getMSLevel()) == *it_mslevel)
+          {
+            int m;    // spectrum index
+
+            // go forward
+            m = n;
+            for (typename MapType::const_iterator it_rt_2 = it_rt; it_rt_2 != exp.end() && std::exp(factor*pow(it_rt_2->getRT() - it_rt->getRT(), 2)) >= cutoff; ++it_rt_2)
+            {
+              if (Int(it_rt_2->getMSLevel()) == *it_mslevel)
+              {                
+                std::pair<Size, double> p(m, std::exp(factor*pow(it_rt_2->getRT() - it_rt->getRT(), 2)));
+                spectra_to_average_over[n].push_back(p);
+              }
+              ++m;
+             }
+            
+            // go backward
+            m = n;
+            for (typename MapType::const_iterator it_rt_2 = it_rt; it_rt_2 != exp.begin() && std::exp(factor*pow(it_rt_2->getRT() - it_rt->getRT(), 2)) >= cutoff; --it_rt_2)
+            {
+              if (Int(it_rt_2->getMSLevel()) == *it_mslevel)
+              {
+                if (m!=n)    // already covered in forward case
+                {
+                  std::pair<Size, double> p(m, std::exp(factor*pow(it_rt_2->getRT() - it_rt->getRT(), 2)));
+                  spectra_to_average_over[n].push_back(p);
+                }
+              }
+              --m;
+            }
+
+          }
+          ++n;
+        }
```

Comment:
most of this code except the `std::exp(factor*pow(it_rt_2->getRT() - it_rt->getRT(), 2))` is copy paste from above. this needs to be changed, best by a parameter given to the function. If this impacts speed too much, there is for sure a template-solution (compile time) that allows through specialization a selection of an appropriate function (e.g. one that returns 1 or one that returns the gaussian). All template uglyness is better than copy-pasted code

See for example this: 
http://stackoverflow.com/questions/1174169/function-passed-as-template-argument